### DISCUSSION TITLE
[Android] Any way I can emit events to JS from a foregound service?

### DISCUSSION THREAD

**Discussion by marcofuentes05** [reactwg/react-native-new-architecture#252] [Deep Dive]
So my app relies on a foreground service to provide data continuously to the JS part. With the old arch I had managed to start the foreground service -> fire up a broadcast to be picked by a receiver -> use the receiver's context to emit a event to JS. However I can't find a way to do this with the new arch since the API to emit events has changed.

**Comment by lovegaoshi** [comment]
u can inherit the service from [HeadlessJsTaskService](https://github.com/robik/react-native/blob/ac870aa5e6a65f4253c2942aa1d8eaab408d1c90/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/HeadlessJsTaskService.java), which has a [reactContext](https://github.com/robik/react-native/blob/ac870aa5e6a65f4253c2942aa1d8eaab408d1c90/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/HeadlessJsTaskService.java#L101). then u can [emitDeviceEvent](https://github.com/lovegaoshi/react-native-track-player/blob/71b08733e6f680db10613036c450509c855917ea/android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt#L777) from it
